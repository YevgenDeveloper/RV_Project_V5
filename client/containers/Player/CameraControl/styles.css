@import 'client/styles/variables';

.control {
  display: flex;
  flex-flow: row nowrap;
  margin-right: 10px;
}

.button {
  margin: 0 1px 0 0;
  padding: 0 10px;
  border-radius: 0;
  white-space: nowrap;
  font-size: 11px;
  text-transform: uppercase;

  &:first-child {
    border-radius: 4px 0 0 4px;
  }

  &:last-child {
    border-radius: 0 4px 4px 0;
    margin-left: 0;
  }

  &.on {
    color: white;
    background-color: $color-blue;

    &:hover:enabled {
      background-color: color-mod($color-blue l(48%));
    }

    &:active:enabled {
      background-color: color-mod($color-blue l(44%));
    }
  }
}
