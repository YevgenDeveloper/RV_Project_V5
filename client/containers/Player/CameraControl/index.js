// @flow

import React from 'react';
import classNames from 'classnames/bind';

import { cameras } from 'client/constants';
import { Button } from 'client/components';
import styles from './styles.css';

const cx = classNames.bind(styles);

type Props = {
  disabled: boolean,
  camera: string,
  onChangeCamera: string => void,
};

const CameraControl = (props: Props) => (
  <div className={styles.control}>
    <Button
      label="Cam 1"
      onClick={() => props.onChangeCamera(cameras.CAMERA_1)}
      className={cx(styles.button, {
        [styles.on]: !props.disabled && props.camera === cameras.CAMERA_1,
      })}
      disabled={props.disabled}
    />
    <Button
      label="Cam 2"
      onClick={() => props.onChangeCamera(cameras.CAMERA_2)}
      className={cx(styles.button, {
        [styles.on]: !props.disabled && props.camera === cameras.CAMERA_2,
      })}
      disabled={props.disabled}
    />
    <Button
      label="Cam 3"
      onClick={() => props.onChangeCamera(cameras.CAMERA_3)}
      className={cx(styles.button, {
        [styles.on]: !props.disabled && props.camera === cameras.CAMERA_3,
      })}
      disabled={props.disabled}
    />
    <Button
      label="Cam 4"
      onClick={() => props.onChangeCamera(cameras.CAMERA_4)}
      className={cx(styles.button, {
        [styles.on]: !props.disabled && props.camera === cameras.CAMERA_4,
      })}
      disabled={props.disabled}
    />
  </div>
);

export default CameraControl;
