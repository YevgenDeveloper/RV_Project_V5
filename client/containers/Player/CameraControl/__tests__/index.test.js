// @flow

import type { ShallowWrapper } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import CameraControl from '..';

it('renders correctly', () => {
  const wrapper: ShallowWrapper = shallow(
    <CameraControl disabled={false} camera="Test" onChangeCamera={jest.fn()} />,
  );

  expect(wrapper).toMatchSnapshot();
});
