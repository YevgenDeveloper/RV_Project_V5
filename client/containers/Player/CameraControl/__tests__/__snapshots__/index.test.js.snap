// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CameraControl
    camera="Test"
    disabled={false}
    onChangeCamera={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          blue={false}
          className="button"
          dark={false}
          disabled={false}
          label="Cam 1"
          onClick={[Function]}
        />,
        <Button
          blue={false}
          className="button"
          dark={false}
          disabled={false}
          label="Cam 2"
          onClick={[Function]}
        />,
        <Button
          blue={false}
          className="button"
          dark={false}
          disabled={false}
          label="Cam 3"
          onClick={[Function]}
        />,
        <Button
          blue={false}
          className="button"
          dark={false}
          disabled={false}
          label="Cam 4"
          onClick={[Function]}
        />,
      ],
      "className": "control",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "blue": false,
          "className": "button",
          "dark": false,
          "disabled": false,
          "label": "Cam 1",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "blue": false,
          "className": "button",
          "dark": false,
          "disabled": false,
          "label": "Cam 2",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "blue": false,
          "className": "button",
          "dark": false,
          "disabled": false,
          "label": "Cam 3",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "blue": false,
          "className": "button",
          "dark": false,
          "disabled": false,
          "label": "Cam 4",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            blue={false}
            className="button"
            dark={false}
            disabled={false}
            label="Cam 1"
            onClick={[Function]}
          />,
          <Button
            blue={false}
            className="button"
            dark={false}
            disabled={false}
            label="Cam 2"
            onClick={[Function]}
          />,
          <Button
            blue={false}
            className="button"
            dark={false}
            disabled={false}
            label="Cam 3"
            onClick={[Function]}
          />,
          <Button
            blue={false}
            className="button"
            dark={false}
            disabled={false}
            label="Cam 4"
            onClick={[Function]}
          />,
        ],
        "className": "control",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": false,
            "className": "button",
            "dark": false,
            "disabled": false,
            "label": "Cam 1",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": false,
            "className": "button",
            "dark": false,
            "disabled": false,
            "label": "Cam 2",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": false,
            "className": "button",
            "dark": false,
            "disabled": false,
            "label": "Cam 3",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": false,
            "className": "button",
            "dark": false,
            "disabled": false,
            "label": "Cam 4",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
